### PR TITLE
Split advanced.md into multiple files + update navigation

### LINKED ISSUE #469: More Structure for advanced.md
advanced.md is quite a collection of headlines right now. This could and should be structure a bit more to offer the user more guidance.

(Will probably only work if #455 is fixed somehow)

### PR DESCRIPTION
**Pull request by janpio**
This Pull Request splits the former `advanced.md` into multiple files. I did my best to move the sections into appropriate files.

We now have:
- /advanced/fastlane/
- /advanced/Fastfile/
- /advanced/Appfile/
- /advanced/lanes/
- /advanced/actions/
- /advanced/other/

The old `/advanced/` is hidden in the menu and redirects to the new file when opened with an anchor.

To get this nice `Fastfile` and `Appfile` file names (that correspond to how they in your projects) I edited the `:ensure_filenames'` lane in the Filename to exclude those both files.

closes #469

### CODE REVIEW

**Comment by janpio** [comment]
via @joshdholtz on Slack: https://5b61933ab3127464aba3f0ca--fastlane-docs-preview.netlify.com/

File: docs/advanced.md [hunk 1]
```diff
@@ -2,1231 +2,6 @@
 
 # Advanced fastlane
 
-# Passing Parameters
+This file is not used any more and its content was moved to subpages. Please follow the links to the content's new location.
```
Comment (getaaron) [inline docs/advanced.md]: Can we do anything better than this? I know I have seen links out there to specific sections of this file using the HTML anchor.

Is it possible to 302 redirect to the correct location?

If not, can we keep the Markdown headings (which generate the anchors) and provide links to the appropriate section underneath?

Comment (janpio) [reply docs/advanced.md]: This PR is still WIP, I was planning on re-adding the headlines here with links to the new locations of the content.

302 redirect does not work, as the anchor is not submitted to the remote server but only handled in the browser. We could write some Javascript that reads this and knows where to redirect.

Comment (janpio) [reply docs/advanced.md]: This looks promising:
https://stackoverflow.com/a/21198129/252627
(only needs an additional check if a new URL was actually found before redirecting)

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>mkdocs.yml</code> was modified - make sure no modification was made by mistake</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">It seems like the page <code>Advanced: advanced.md</code> got removed from mkdocs.yml, we don’t want to remove any public links. Instead please add a redirect to the new page.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by janpio** [comment]
So I did some work on redirecting the old headlines/anchors to the new pages.

A list of all anchors:
https://docs.fastlane.tools/advanced/#passing-parameters
https://docs.fastlane.tools/advanced/#switching-lanes
https://docs.fastlane.tools/advanced/#output-environment-variables
https://docs.fastlane.tools/advanced/#interacting-with-the-user
https://docs.fastlane.tools/advanced/#run-actions-directly
https://docs.fastlane.tools/advanced/#shell-values
https://docs.fastlane.tools/advanced/#priorities-of-parameters-and-options
https://docs.fastlane.tools/advanced/#importing-another-fastfile
https://docs.fastlane.tools/advanced/#using-fastlane_require
https://docs.fastlane.tools/advanced/#environment-variables
https://docs.fastlane.tools/advanced/#lane-properties
https://docs.fastlane.tools/advanced/#lane-context
https://docs.fastlane.tools/advanced/#private-lanes
https://docs.fastlane.tools/advanced/#load-own-actions-from-external-folder
https://docs.fastlane.tools/advanced/#skip-update-check-when-launching-fastlane
https://docs.fastlane.tools/advanced/#hide-changelog-information-at-the-end-of-running-fastlane
https://docs.fastlane.tools/advanced/#adding-credentials
https://docs.fastlane.tools/advanced/#control-configuration-by-lane-and-by-platform
https://docs.fastlane.tools/advanced/#manually-manage-the-fastlane-match-repo
https://docs.fastlane.tools/advanced/#directory-behavior
https://docs.fastlane.tools/advanced/#appfile
https://docs.fastlane.tools/advanced/#building-actions

(Ugh that will be quite some work to list...)

Anchors with the netlify preview hostname:
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#passing-parameters
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#switching-lanes
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#output-environment-variables
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#interacting-with-the-user
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#run-actions-directly
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#shell-values
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#priorities-of-parameters-and-options
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#importing-another-fastfile
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#using-fastlane_require
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#environment-variables
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#lane-properties
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#lane-context
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#private-lanes
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#load-own-actions-from-external-folder
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#skip-update-check-when-launching-fastlane
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#hide-changelog-information-at-the-end-of-running-fastlane
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#adding-credentials
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#control-configuration-by-lane-and-by-platform
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#manually-manage-the-fastlane-match-repo
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#directory-behavior
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#appfile
https://deploy-preview-654--fastlane-docs-preview.netlify.com/advanced/#building-actions

Currently a problem is that the old `advanced.md` has to be in the navigation so the page gets actually generated. Probably putting them into a `_` navigation item would make this invisible. Will have to try.

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>mkdocs.yml</code> was modified - make sure no modification was made by mistake</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">It seems like the page <code>Advanced: advanced.md</code> got removed from mkdocs.yml, we don’t want to remove any public links. Instead please add a redirect to the new page.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: docs/advanced.md (hunk 1 above)
Comment (janpio) [reply docs/advanced.md]: This is now implemented and works pretty well.

File: docs/advanced/other.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+Credential felix@krausefx.com:********* added to keychain.
+```
+
+**Removing a Credential**
+```no-highlight
+fastlane fastlane-credentials remove --username felix@krausefx.com
+password has been deleted.
+```
+
+# Manually manage the fastlane match Repo
```
Comment (KrauseFx) [inline docs/advanced/other.md]: This should probably be part of the `match.md` README in the `fastlane/fastlane` repo instead, so it's generated as part of `actions/match`

File: mkdocs.yml [hunk 3]
```diff
@@ -262,4 +262,13 @@ pages:
   - Source Control: best-practices/source-control.md
   - Continuous Integration: best-practices/continuous-integration.md
   - Keys: best-practices/keys.md
-- Advanced: advanced.md
+- Advanced:
```
Comment (KrauseFx) [inline mkdocs.yml]: Can we make this behave similarly to how the `Actions` buttons behaves? Folded by default, and when clicked, it folds up, and shows an overview of all sub-pages basically?

Comment (KrauseFx) [reply mkdocs.yml]: I think with our current system this should work without extra effort, right?

File: docs/advanced/other.md (hunk 2 above)
Comment (janpio) [reply docs/advanced/other.md]: Good point, let's create an issue for it though and get this merged here so the changes are not mixed?

File: mkdocs.yml (hunk 3 above)
Comment (janpio) [reply mkdocs.yml]: That's actually quite a bit of extra, hacky work as "Actions" has a sister "_Actions" that holds the action pages and there are multiple places where this is configured to work... But it is possible, sure. Worth the effort?

**Comment by janpio** [comment]
Will do, but have to fix that pesky conflict first :/

**Pull request merged**